src/Kestrel.Core/Internal/Http/Http1MessageBodyPipeReader.cs

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
+{
+    public class Http1MessageBodyPipeReader : PipeReader, IAwaiter<ReadResult>
```

Comment:
Where do we throw                      BadHttpRequestException.Throw(RequestRejectionReason.UnexpectedEndOfRequestContent);
 now?